osbuild-run

```diff
@@ -5,25 +5,66 @@ import subprocess
 import sys
 
 
-# ld.so.conf must exist, or `ldconfig` throws a warning
-subprocess.run(["touch", "/etc/ld.so.conf"], check=True)
-subprocess.run(["ldconfig"], check=True)
-
-try:
-    subprocess.run(["systemd-sysusers"], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=True)
-except subprocess.CalledProcessError as error:
-    sys.stderr.write(error.stdout)
-    sys.exit(1)
-
-# generate /etc/pki/tls/certs/ca-bundle.crt
-os.makedirs("/etc/pki/ca-trust/extracted/pem")
-os.makedirs("/etc/pki/tls/certs")
-os.symlink("/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem", "/etc/pki/tls/certs/ca-bundle.crt")
-subprocess.run(["update-ca-trust"])
-
-# Allow systemd-tmpfiles to return non-0. Some packages want to create
-# directories owned by users that are not set up with systemd-sysusers.
-subprocess.run(["systemd-tmpfiles", "--create"])
-
-r = subprocess.run(sys.argv[1:])
-sys.exit(r.returncode)
+def osbuild_setup_ldconfig():
+    # ld.so.conf must exist, or `ldconfig` throws a warning
+    subprocess.run(["touch", "/etc/ld.so.conf"], check=True)
+    subprocess.run(["ldconfig"], check=True)
+
+
+def osbuild_setup_sysusers():
+    try:
+        subprocess.run(["systemd-sysusers"], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=True)
+    except subprocess.CalledProcessError as error:
+        sys.stderr.write(error.stdout)
+        sys.exit(1)
+
+
+def append_certs(cert_conf, dir_fd, parents=b""):
+    for entry in os.scandir(f"/proc/self/fd/{dir_fd}".encode()):
+        if entry.is_file():
+            line = os.path.join(parents, entry.name)
+            print(f"adding: {line}")
+            cert_conf.write(line)
+            cert_conf.write(b"\n")
+        elif entry.is_dir():
+            append_certs(cert_conf, os.open(entry.name, os.O_DIRECTORY, dir_fd=dir_fd), os.path.join(parents, entry.name))
+
+
+def osbuild_setup_certs():
+    # generate /etc/pki/tls/certs/ca-bundle.crt
+    os.makedirs("/etc/pki/ca-trust/extracted/pem")
+    os.makedirs("/etc/pki/tls/certs")
+    os.symlink("/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem", "/etc/pki/tls/certs/ca-bundle.crt")
+
+    try:
+        subprocess.run(["update-ca-trust"])
+    except FileNotFoundError:
+        pass
+
+    os.makedirs("/etc/ssl/certs")
+    with open("/etc/ca-certificates.conf", "wb") as f:
+        append_certs(f, os.open("/usr/share/ca-certificates", os.O_DIRECTORY))
+        #try:
+        #except FileNotFoundError:
+        #    pass
+    subprocess.run(["cat", "/etc/ca-certificates.conf"])
+    subprocess.run(["update-ca-certificates"])
+    subprocess.run(["cat", "/etc/ssl/certs/ca-certificates.crt"])
+
+
+def osbuild_setup_tmpfiles():
+    # Allow systemd-tmpfiles to return non-0. Some packages want to create
+    # directories owned by users that are not set up with systemd-sysusers.
+    subprocess.run(["systemd-tmpfiles", "--create"])
+
+
+def osbuild_setup():
+    osbuild_setup_ldconfig()
+    osbuild_setup_sysusers()
+    osbuild_setup_certs()
```

Comment:
I appreciate the cleanup, but I don't think we need this function, nor the `osbuild_` prefixes.

Reply:
Ignore the prefix: I used to be a C programmer...
If you meant to call them all from `if __name__ == "__main__"`, then I agree. Or did you mean not to wrap them in helper functions at all?

Reply:
I think one level of helper functions is enough. I.e., call `setup_*()` from `main`, yeah. (Don't feel strongly about it though.)

Reply:
Agreed, changed.